iree/turbine/kernel/wave/decompose_vmma_ops.py

```diff
@@ -0,0 +1,166 @@
+# Copyright 2024 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+from ..wave.constraints import (
+    Constraint,
+    HardwareConstraint,
+    MMAType,
+    MMAOperand,
+)
+from iree.turbine.kernel.wave.utils import get_mfma_load_elems_per_thread
+from .._support.tracing import CapturedTrace
+from .._support.indexing import IndexSequence, IndexSymbol, IndexExpr
+from ..ops.wave_ops import (
+    get_custom,
+    MMA,
+    Reshape,
+)
+from ..lang.global_symbols import *
+
+import copy
+import sympy
+
+
+VMMA_TO_NATIVE_MAP = {
+    MMAType.F32_16x16x32_K8_F16: MMAType.F32_16x16x16_F16,
+    MMAType.F32_32x32x16_K8_F16: MMAType.F32_32x32x8_F16,
+}
+
+
+def get_m_dim(mma_op):
+    m_dims = set(mma_op.lhs_type.symbolic_shape).difference(
+        mma_op.rhs_type.symbolic_shape
+    )
+    assert len(m_dims) == 1, "Expect to have single M-dim."
+    return m_dims.pop()
+
+
+def get_n_dim(mma_op):
+    n_dims = set(mma_op.rhs_type.symbolic_shape).difference(
+        mma_op.lhs_type.symbolic_shape
+    )
+    assert len(n_dims) == 1, "Expect to have single N-dim."
+    return n_dims.pop()
+
+
+def get_k_dim(mma_op):
+    k_dims = set(mma_op.lhs_type.symbolic_shape).intersection(
```

Comment:
Could use the mma_op's reduction dim here ?

Reply:
makes sense, done, thanks!